Framework/Core/src/ConfigParamsHelper.cxx

```diff
@@ -22,10 +22,13 @@ namespace framework {
 /// taking the VariantType into account
 void populateBoostProgramOptions(
     bpo::options_description &options,
-    const std::vector<ConfigParamSpec> &specs
+    const std::vector<ConfigParamSpec> &specs,
+    bpo::options_description vetos
   ) {
   auto proxy = options.add_options();
   for (auto & spec : specs) {
+    // skip everything found in the veto definition
+    if (vetos.find_nothrow(spec.name, false)) continue;
```

Comment:
this filter was missing in the first version. the program option definitions from this function are added as custom options to the device. Since FairMQDevice already defines e.g. `--channel-config`, this definition would conflict with a definition from the config spec.